tests/tests_caching_logic.py

```diff
@@ -84,6 +84,8 @@ async def test_num_infer_calls(self):
       del gt_engine
       del aidb_engine
     p.terminate()
+    p.join()
```

Comment:
Also, I cannot terminate the test server completely without this. I'm not sure whether it's a bug or I'm not using the correct way to write multiple tests.

Reply:
Please check this in depth

Reply:
I believe the test server wasn't properly closed before. The server will still occupy the port if only the terminate() function is called. However most test classes only have one unit test or the service is not called, so it did not cause any problems.